src/OSPSuite.Presentation/Presenters/Importer/ImporterDataPresenter.cs

```diff
@@ -257,10 +250,10 @@ public void RefreshTabs()
 
       public void DisableImportedSheets()
       {
-         if (Sheets.Keys.Any(x => x == View.SelectedTab))
+         if (ImportedSheets.Any(x => x.SheetName == View.SelectedTab))
             View.DisableImportCurrentSheet();
 
-         if (Sheets.Keys.All(GetSheetNames().Contains) && GetSheetNames().Count == Sheets.Keys.Count())
+         if (ImportedSheets.All(sheet => GetSheetNames().Contains(sheet.SheetName)) && GetSheetNames().Count == ImportedSheets.Count())
```

Comment:
GetSheetNames() is called twice. You probably want to cash it before and do the count first as opposed to the check

Reply:
We also have a COntainsAll method. Not sure if this is what you need. TO investigate as it might not check for the count

Reply:
done